src/frontend/apps/impress/src/features/language/LanguagePicker.tsx

```diff
@@ -33,43 +39,62 @@ const SelectStyled = styled(Select)<{ $isSmall?: boolean }>`
 
 export const LanguagePicker = () => {
   const { t, i18n } = useTranslation();
-  const { preload: languages } = i18n.options;
+  const { toast } = useToastProvider();
+  const { mutateAsync: changeUserLanguage } = useChangeUserLanguage();
+  const { userData } = useAuthStore();
 
-  const optionsPicker = useMemo(() => {
-    return (languages || []).map((lang) => ({
-      value: lang,
-      label: lang,
-      render: () => (
-        <Box
-          className="c_select__render"
-          $direction="row"
-          $gap="0.7rem"
-          $align="center"
-        >
-          <Text $isMaterialIcon $size="1rem" $theme="primary" $variation="600">
-            translate
-          </Text>
-          <Text $theme="primary" $variation="600">
-            {LANGUAGES_ALLOWED[lang]}
-          </Text>
-        </Box>
-      ),
-    }));
-  }, [languages]);
+  const optionsPicker = Object.keys(LANGUAGES_ALLOWED).map((lang) => ({
+    value: lang,
+    label: lang,
+    render: () => (
+      <Box
+        className="c_select__render"
+        $direction="row"
+        $gap="0.7rem"
+        $align="center"
+      >
+        <Text $isMaterialIcon $size="1rem" $theme="primary" $variation="600">
+          translate
+        </Text>
+        <Text $theme="primary" $variation="600">
+          {LANGUAGES_ALLOWED[lang]}
+        </Text>
+      </Box>
+    ),
+  }));
 
   return (
     <SelectStyled
       label={t('Language')}
       showLabelWhenSelected={false}
       clearable={false}
       hideLabel
-      defaultValue={i18n.language}
+      value={
+        Object.keys(LANGUAGES_ALLOWED).find(
```

Comment:
Should we use `LANGUAGES` of the conf instead ?
```
const { data: conf } = useConfig();
```

Reply:
✅